components/sidebar/sidebar_category/sidebar_category.tsx

```diff
@@ -157,33 +153,32 @@ export default class SidebarCategory extends React.PureComponent<Props, State> {
 
         if (category.type === CategoryTypes.DIRECT_MESSAGES) {
             return draggingState.type === DraggingStateTypes.CHANNEL;
-        } else if (category.type === CategoryTypes.PUBLIC || category.type === CategoryTypes.PRIVATE) {
+        } else if (category.type === CategoryTypes.CHANNELS) {
             return draggingState.type === DraggingStateTypes.DM;
         }
 
         return false;
     }
 
     render() {
-        const {category, categoryIndex, isCollapsed, draggingState, channels} = this.props;
-        const {isNewCategory} = this.state;
+        const {category, categoryIndex, isCollapsed, draggingState, channels, isNewCategory} = this.props;
 
         if (!category) {
             return null;
         }
 
-        if (!isNewCategory && category.type !== CategoryTypes.DIRECT_MESSAGES && (!channels || !channels.length)) {
+        if (!isNewCategory && category.type !== CategoryTypes.DIRECT_MESSAGES && category.type !== CategoryTypes.CUSTOM && (!channels || !channels.length)) {
```

Comment:
we need `!isNewCategory && !(channels || channels.length)` since `isNewCategory` is view state that isn't reset until refreshed, so we check if channels have been dragged in to clear this state.

Reply:
Shouldn't `isNewCategory` always be false for non-custom categories? The only categories that are hidden when empty are Channels and Favorites, so we don't need to check `isNewCategory` for them.

Reply:
Ahh good point. Will rework.